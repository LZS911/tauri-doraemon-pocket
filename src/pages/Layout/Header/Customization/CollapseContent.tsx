import { CollapseContentProps } from '.';
import usePropsValue from '../../../../hooks/usePropsValue';

const CollapseContent: <ValueType>(
  props: CollapseContentProps<ValueType>
) => React.ReactElement = ({ value, defaultValue, onChange, list }) => {
  const [internalValue, setInternalValue] = usePropsValue({
    value,
    defaultValue,
    onChange,
  });
  return (
    <div className="flex flex-wrap">
      {list.map((v) => {
        return (
          <div
            onClick={() => {
              setInternalValue(v.value);
            }}
            key={v.key}
            className={`mr-2 flex cursor-pointer flex-col items-center rounded px-4 py-4 pb-1 text-center transition-[background] ${
              v.value === internalValue ? 'bg-secondary' : ''
            }`}
          >
            <div className="mb-1">{v.content && v.content}</div>
            {v.describe && (
              <span className="text-xs opacity-80">{v.describe}</span>
            )}
          </div>
        );
      })}
    </div>
  );
};

export default CollapseContent;
