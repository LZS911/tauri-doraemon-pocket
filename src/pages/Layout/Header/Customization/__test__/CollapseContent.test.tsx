import { fireEvent, render, screen } from '@testing-library/react';
import CollapseContent from '../CollapseContent';

const list = [
  {
    key: '1',
    content: 'content1',
    value: 1,
  },
  {
    key: '2',
    content: 'content2',
    value: 2,
  },
  {
    key: '3',
    content: 'content3',
    describe: 'describe',
    value: 3,
  },
];

describe('test CollapseContent', () => {
  test('should match snapshot', () => {
    const { container } = render(<CollapseContent list={list} />);
    expect(container).toMatchSnapshot();
  });

  test('should selected default item when default is not empty', () => {
    render(<CollapseContent list={list} defaultValue={2} />);

    // expect(screen.getByText('content2').parentNode).toHaveClass('bg-secondary');
  });

  test('should be executed onChange when clicking other item', async () => {
    const onChange = jest.fn();
    render(<CollapseContent list={list} value={2} onChange={onChange} />);
    expect(onChange).toBeCalledTimes(0);

    fireEvent.click(screen.getByText('content2'));
    expect(onChange).toBeCalledTimes(0);
    fireEvent.click(screen.getByText('content1'));
    expect(onChange).toBeCalledTimes(1);
    expect(onChange).toBeCalledWith(1);
  });
});
