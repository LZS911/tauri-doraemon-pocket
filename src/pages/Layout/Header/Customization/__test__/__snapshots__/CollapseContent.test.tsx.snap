// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CollapseContent should match snapshot 1`] = `
<div>
  <div
    class="flex flex-wrap"
  >
    <div
      class="mr-2 flex cursor-pointer flex-col items-center rounded px-4 py-4 pb-1 text-center transition-[background] "
    >
      <div
        class="mb-1"
      >
        content1
      </div>
    </div>
    <div
      class="mr-2 flex cursor-pointer flex-col items-center rounded px-4 py-4 pb-1 text-center transition-[background] "
    >
      <div
        class="mb-1"
      >
        content2
      </div>
    </div>
    <div
      class="mr-2 flex cursor-pointer flex-col items-center rounded px-4 py-4 pb-1 text-center transition-[background] "
    >
      <div
        class="mb-1"
      >
        content3
      </div>
      <span
        class="text-xs opacity-80"
      >
        describe
      </span>
    </div>
  </div>
</div>
`;
